_docs/technical/2017-01-13-blocks.md

```diff
@@ -11,160 +11,130 @@ group: technical
 
 ## Block Application
 
-We apply blocks using the [`applyBlocks`](https://github.com/input-output-hk/cardano-sl/blob/517a72801c0bbb11a34c8d6a6d528fff5f094471/src/Pos/Block/Logic.hs#L474)
+We apply blocks using the [`applyBlocks`](https://github.com/input-output-hk/cardano-sl/blob/09e4fcf8a7f1a5a587d0241aa93f23e2d72c7c70/src/Pos/Block/Logic/VAR.hs#L182)
 function. The sequence of blocks should be definitely valid:
 we must verify all predicates and data checks regarding blocks.
+
 **Important**: all blocks in that sequence must be of the same epoch!
 
-If all conditions are met, we [actually apply blocks](https://github.com/input-output-hk/cardano-sl/blob/517a72801c0bbb11a34c8d6a6d528fff5f094471/src/Pos/Block/Logic/Internal.hs#L77):
+If all conditions are met, we [actually apply blocks](https://github.com/input-output-hk/cardano-sl/blob/09e4fcf8a7f1a5a587d0241aa93f23e2d72c7c70/src/Pos/Block/Logic/Internal.hs#L107):
 
-* [apply US (Update System)](https://github.com/input-output-hk/cardano-sl/blob/517a72801c0bbb11a34c8d6a6d528fff5f094471/src/Pos/Update/Logic/Global.hs#L60),
-* [apply delegation](https://github.com/input-output-hk/cardano-sl/blob/517a72801c0bbb11a34c8d6a6d528fff5f094471/src/Pos/Delegation/Logic.hs#L290),
-* [apply transactions](https://github.com/input-output-hk/cardano-sl/blob/517a72801c0bbb11a34c8d6a6d528fff5f094471/src/Pos/Txp/Logic.hs#L72).
+* [apply US (Update System)](https://github.com/input-output-hk/cardano-sl/blob/09e4fcf8a7f1a5a587d0241aa93f23e2d72c7c70/src/Pos/Block/Logic/Internal.hs#L141),
+* [apply delegation](https://github.com/input-output-hk/cardano-sl/blob/09e4fcf8a7f1a5a587d0241aa93f23e2d72c7c70/src/Pos/Block/Logic/Internal.hs#L142),
+* [apply transactions](https://github.com/input-output-hk/cardano-sl/blob/09e4fcf8a7f1a5a587d0241aa93f23e2d72c7c70/src/Pos/Block/Logic/Internal.hs#L143).
 
 Moreover, we can verify blocks before application (i.e. apply blocks only if
-they're valid). We use [`verifyAndApplyBlocks`](https://github.com/input-output-hk/cardano-sl/blob/517a72801c0bbb11a34c8d6a6d528fff5f094471/src/Pos/Block/Logic.hs#L404)
+they're valid). We use [`verifyAndApplyBlocks`](https://github.com/input-output-hk/cardano-sl/blob/11fdc35884368ba61ec32e3277c037e20dc1c266/src/Pos/Block/Logic/VAR.hs#L99)
 function for it. If some error occurred during application, there are two options:
 
-1. All blocks applied inside this function will be rollbacked.
-2. This function will try to apply as many blocks as possible.
+1. All blocks applied inside this function will be [rollbacked](https://github.com/input-output-hk/cardano-sl/blob/11fdc35884368ba61ec32e3277c037e20dc1c266/src/Pos/Block/Logic/VAR.hs#L137).
```

Comment:
"rollbacked" -> "rolled back"

I get why you did this, but I'd say it's better this way.